### ISSUE TITLE
Rehabilitate or remove PSIO multivolume striping functionality

### ISSUE THREAD

**Issue by TiborGY** [psi4/psi4#2725]
Poking around the PSIO machinery, it seems to me that it is meant to be capable of doing a sort of "userland RAID0", where IO is striped over multiple scratch directories. But as far as I can tell, this is not (in any shape or form) accessible to users, so the number of volumes is always 1.

While this is a somewhat interesting feature, and I can imagine that there are circumstances where it would be useful, it comes at the cost of increasing code complexity in PSIO.

Right now, it is the worst of both worlds: noone can benefit from the feature, and PSIO code is more complicated. So IMO, this feature should be either rehabilitated and documented, or stripped out.

I am not volunteering for either, so this may very well be one of those "for the sake of posterity" issues.

**Comment by lothian** [comment]
Yes, I wrote the original version of `libpsio`, and this I/O structure was an important part of its use for many years.  Indeed, it was intended to replicate some of the key performance features of an earlier I/O library (long since removed) contained within the old `libciomr` that have infrastructure for about four different types of I/O.   However, at some point in the development of Psi4, someone wrote the new, much more complicated code for building the file paths, they broke this capability.  Chesterton's fence strikes again.

**Comment by JonathonMisiewicz** [comment]
The "correct" solution is probably to replace PSIO with hdf5. I'd like to talk about that at PsiCon.

**Comment by TiborGY** [comment]
> The "correct" solution is probably to replace PSIO with hdf5. I'd like to talk about that at PsiCon.

Q5Cost ?
I have seen some evidence of Q5Cost support in the MRCC codebase, so that could maybe replace the currently very fragile MRCC interface.

**Comment by JonathonMisiewicz** [comment]
Further discussion of replacing `libpsio` should be moved to a new issue. As for your specific proposal:

I'm not familiar with Q5Cost, and a quick Google search gives me a 14-year-old manual with so many out-of-date links that I doubt Q5Cost is actively maintained. That's a compelling argument against adopting it.

Q5Cost was a proposed standard format for tensors-with-metadata. If MRCC is the only other major user and we have no reason to believe in other users, interoperability is not a compelling reason.

**Comment by TiborGY** [comment]
Closed by #3452 : the PSIO multivolume striping functionality is removed.
